Source/WebKit2/WebProcess/WebPage/WebPage.cpp

```diff
@@ -2704,6 +2704,25 @@ void WebPage::setUserAgent(const String& userAgent)
     m_userAgent = userAgent;
 }
 
+void WebPage::applyRequestHeaders(WebFrame* frame, ResourceRequest& request) const
+{
+    if (!frame || !m_loaderClient.client().requestHeadersForURL)
+        return;
+
+    API::Dictionary* headers = m_loaderClient.requestHeadersForURL(frame, API::URL::create(request.url()).ptr());
+    if (!headers)
+        return;
+
+    for (auto& item : headers->map()) {
+        if (item.value.get()->type() != API::String::APIType)
+            continue;
+
+        API::String* value = item.value.get();
+        request.setHTTPHeaderField(item.key, value->string());
+    }
+    headers->deref();
```

Comment:
Decided to deref here, just because could not find more proper way to destroy API::Dictionary returned from injected bundle